### DISCUSSION TITLE
Confusion about Window width and Center and Hounsfield units

### DISCUSSION THREAD

**Discussion by soothingmusic** [fo-dicom/fo-dicom#1438] [General]
I am currently adjusting the brightness and contrast of a grayscale image using fo-dicom.

Loading the image:

    _dicomFile = DicomFile.Open(GetImageFileName());
    _dicomImage = new DicomImage(_dicomFile.Dataset);
    WindowLevel = _dicomImage.WindowCenter;
    WindowWidth = _dicomImage.WindowWidth;
    var grayScaleOptions = GrayscaleRenderOptions.FromBitRange(_dicomFile.Dataset);
    Depth = grayScaleOptions.BitDepth.BitsAllocated;

    DisplayedImage = _dicomImage.RenderImage().As<WriteableBitmap>();
On mouse move, I am adjusting the brightness and contrast

 public void OnMouseMove(object sender, MouseCaptureArgs e)
    {
       //...
        _dicomImage.WindowWidth += e.X - _lastXPos;
        _dicomImage.WindowCenter += e.Y - _lastYPos;
        _dicomImage.UseVOILUT = false;

        _lastXPos = e.X;
        _lastYPos = e.Y;
        DisplayedImage = _dicomImage.RenderImage().As<WriteableBitmap>();
    }
Everything is working as expected. I am little confused where Hounsfield values come into picture. I know about how HU is calculated based on slope and intercept.

I have checked the fo-dicom code and did not find any references to HU.

If I display the WW and WC values that I am calculating here on the image, it won't be the HU values that user expects to see?

**Comment by gofal** [comment]
Sure, the WindowWidth and WindowCenter are in HU base.

in DICOM there is a render pipeline. First the uncompressed pixel are read from file, then some conversions are done, like some linear transformation via Intercept/rescale. there may also be some ModalityLUT to lookup the pixel data.
This results in the so called p-value. In case of CT this p-value has a unit called Hounsfield. For other modalities this p-value has no unit.
And then finally this p-value (in case of CT some value from -1024 up to several thouthands) has to be transformed into a gray-value between 0 and 255 that can be displayed on a screen. for that last transformation you are doing WindowCenter/WindowWidth calcualtion or you could use a VOILut.
